### PR TITLE
jiten: cherry-pick test fix from upstream

### LINKED ISSUE #271600: Build failure: jiten
### Steps To Reproduce

1. Check out nixos-23.11 and cd into it
2. NIXPKGS_ALLOW_UNFREE nix-build -A jiten

### Build log

https://gist.github.com/IreneKnapp/8c1284536cc214e5736a6b06e7f443d2

### Additional context

This same version of jiten works fine on 23.05.

### Notify maintainers

@obfusk

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.9, NixOS, 23.05 (Stoat), 23.05.20231027.1a3c95e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.6`
 - channels(root): `"nixos-22.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Note that that channel is unused because my system is flake-ified, which is why it's out of date. The host is nixos-23.05 because this came to light while upgrading, but the affected build is nixos-23.11 as described above.

### Priorities

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by IreneKnapp**
## Description of changes

The build of jiten is broken due to a test failure.

This fixes https://github.com/NixOS/nixpkgs/issues/271600 by cherry-picking an unreleased patch from upstream, as discussed on the issue thread.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Whovian9369** [comment]
Appears to build successfully on my end, also on `x86_64-linux`.

**Review by obfusk — APPROVED** [review]
As mentioned in the issue, I can't currently test the package myself, but since others have confirmed my patch seems to work correctly, this PR LGTM. Thanks!

File: pkgs/applications/misc/jiten/default.nix [hunk 1]
```diff
@@ -27,6 +27,10 @@ python3.pkgs.buildPythonApplication rec {
     sha256 = "16sz8i0sw7ggy6kijcx4qyl2zr6xj789x4iav0yyllx12dfgp5b1";
   };
 
+  patches = [
+    ./cookie-fix.patch
```
Comment (Mic92) [inline pkgs/applications/misc/jiten/default.nix:31]:
```suggestion
    # Potentially can be dropped after the next release
    ./cookie-fix.patch
```

Comment (IreneKnapp) [reply pkgs/applications/misc/jiten/default.nix:31]: thanks, good thought. done.

**Review by obfusk — APPROVED** [review]

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 274349` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>jiten</li>
    <li>jiten.dist</li>
  </ul>
</details>

Patch LGTM and tried jiten with `--help` and `jmdict`. Thanks you for the support from upstream!

**Pull request merged**